### DISCUSSION TITLE
Open OAuth in popup window

### DISCUSSION THREAD

**Discussion by DiegoGallegos4** [nuxt-community/auth-module#1583] [Ideas & Feature Requests]
### What problem does this feature solve?

Provide popup login for facebook and google schemes.

### What does the proposed changes look like?

On oauth2.js change the current behavior of window.location to redirect the user to oauth login page. Instead provide options so that it can open a popup to verify and prevent any redirect during login.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/auth-module/issues/c271">#c271</a>)</em></sub></div>

**Comment by DevBrent** [comment]
I imagine you'd have your success URLs set to try to close the new tab? I think you could easily emulate this by having your /login URL auto-forward into the standard oauth process then ensure all links to /login are target="_blank" or triggered by Javascript if you want to try to close the popup window instance after.

**Comment by tomkochi** [comment]
Is this issue solved? Any ideas how to do it? @DevBrent do you know how to add a target=_blank to it?

**Comment by rasgo-cc** [comment]
Looking for the same behavior.

**Comment by steklopod** [comment]
Will be nice

**Comment by ctwhome** [comment]
need it

**Comment by dajpes** [comment]
Do it.

**Comment by mrleblanc101** [comment]
I'd love to see a way to do something like this for all scheme.
This should be possible using `window.open` and the postMessage API.
![login-popup](https://user-images.githubusercontent.com/17392251/109337561-ff308580-7832-11eb-8bcc-1b658a36d8ce.jpg)

**Comment by Red-Asuka** [comment]
need it

**Comment by hamzamogni** [comment]
need it

**Comment by SocketSomeone** [comment]
need it
